test/e2e/fleet_guard_rail_test.go

```diff
@@ -44,13 +46,14 @@ var (
 	testGroups = []string{"system:authenticated"}
 )
 
-var _ = Describe("fleet guard rail tests for deny MC CREATE operations", func() {
+var _ = Describe("fleet guard rail tests for deny fleet MC CREATE operations", func() {
```

Comment:
are those tests duplicate of the functional test in fletresourcehandler? Do we need to repeat all the cases here too?

Reply:
We definitely need to have all the deny cases tested since that's guard rail feature's primary purpose. But there are some cases where allowing a request is equally important for system users to ensure we don't break existing functionalities. We currently have 22 IT specs in E2E for v1beta1 MC compared to 27 UTs.

One thing to note: For the requests in UT, we are mocking the requests so if the actual request behavior changes the E2Es are the only way to catch it

Reply:
do we need the UT then?

Reply:
Addressed https://github.com/Azure/fleet/pull/892